pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -323,6 +322,92 @@ public enum Type {
+       */
+
+      public enum Type {
+        // Adaptive Selectors are not used. Use default round-robin.
+        NO_OP,
+
+        // Use selector that routes to servers based on number of inflight requests.
+        NUM_INFLIGHT_REQ,
+
+        // Use selector that routes to servers based on observed latencies.
+        LATENCY,
+
+        // Uses both latency and numInFlightRequests to select the best server based on C3 score. It is based on the
+        // approach described in https://www.usenix.org/system/files/conference/nsdi15/nsdi15-paper-suresh.pdf
+        HYBRID
+      }
+
+      // Determines the type of AdaptiveServerSelector to use.
+      public static final String CONFIG_OF_TYPE = "pinot.broker.adaptive.server.selector.type";
+      public static final String DEFAULT_TYPE = Type.NO_OP.name();
+
+      public static final String CONFIG_OF_ENABLE_STATS_COLLECTION = "pinot.broker.adaptive.server.selector.enable"
+          + ".stats.collection";
+      public static final boolean DEFAULT_ENABLE_STATS_COLLECTION = false;
+
+      // Parameters to tune exponential moving average.
+
+      // The weightage to be given for a new incoming value. For example, alpha=0.30 will give 30% weightage to the
+      // new value and 70% weightage to the existing value in the Exponential Weighted Moving Average calculation.
+      public static final String CONFIG_OF_ALPHA = "pinot.broker.adaptive.server.selector.alpha";
+      public static final double DEFAULT_ALPHA = 0.666;
+
+      // If the EMA average has not been updated during a specified time window (defined by this property), the
+      // EMA average value is automatically decayed by an incoming value of zero. This is required to bring a server
+      // back to healthy state gracefully after it has experienced some form of slowness.
+      public static final String CONFIG_OF_AUTODECAY_WINDOW_MS =
+          "pinot.broker.adaptive.server.selector.autodecay.window.ms";
+      public static final long DEFAULT_AUTODECAY_WINDOW_MS = 10 * 1000;
+
+      // Determines the initial duration during which incoming values are skipped in the Exponential Moving Average
+      // calculation. Until this duration has elapsed, average returned will be equal to AVG_INITIALIZATION_VAL.
+      public static final String CONFIG_OF_WARMUP_DURATION_MS = "pinot.broker.adaptive.server.selector.warmup.duration";
+      public static final long DEFAULT_WARMUP_DURATION_MS = 0;
+
+      // Determines the initialization value for Exponential Moving Average.
+      public static final String CONFIG_OF_AVG_INITIALIZATION_VAL = "pinot.broker.adaptive.server.selector.avg"
+          + ".initialization.val";
+      public static final double DEFAULT_AVG_INITIALIZATION_VAL = 1.0;
+
+      // Parameters related to C3 score.
+      public static final String CONFIG_OF_C3_SCORE_EXPONENT = "pinot.broker.adaptive.server.selector.c3.score"
```

Comment:
I don't think we should expose the name "C3" as part of public constant.

Reply:
Done